# String: Topic prefix to be prepended to plugin name to find trajectory planning
# service. Only needed for minimal/mock plan delegator
# Units: N/a
planning_topic_prefix: /guidance/plugins/

# String: Topic suffix to be appended to plugin name to find trajectory planning
# service. Only needed for minimal/mock plan delegator
# Units: N/a
planning_topic_suffix: /plan_trajectory

# Double: The trajectory planning frequency
# Units: Hz
trajectory_planning_rate: 10.0

# Double: Plan delegator will stop generating new trajectories
# if the current plan is already over this threshold
# Units: Second
trajectory_duration_threshold: 6.0

# Minimum crawl speed
# Double: The minimum speed the vehicle will say it is moving in order to support accelerations
# Units: m/s
min_speed: 2.2352

# Double: If an upcoming lane change will begin in under this time threshold, a turn signal activation command will be published.
# Units: Seconds
duration_to_signal_before_lane_change: 2.5

# Int: The maximum duration that Plan Delegator will wait after calling a tactical plugin's trajectory planning service; if trajectory 
# generation takes longer than this, then trajectory planning will immediately end for the current trajectory planning iteration.
# NOTE: It is highly desirable to maintain a timeout of 100 ms or less, but trajectory generation success cannot be guaranteed with this duration
#       for tactical plugins (primarily cooperative_lanechange) in all test scenarios at this time.
# Units: Milliseconds
tactical_plugin_service_call_timeout: 100